src/views/preview/preview.jsx

```diff
@@ -97,8 +102,35 @@ class Preview extends React.Component {
     removeEventListeners () {
         window.removeEventListener('popstate', this.handlePopState);
     }
+    getExtensions (projectId) {
+        storage
+            .load(storage.AssetType.Project, projectId, storage.DataFormat.JSON)
+            .then(projectAsset => {
+                let input = projectAsset.data;
+                if (typeof input === 'object' && !(input instanceof ArrayBuffer) &&
+                !ArrayBuffer.isView(input)) { // taken from scratch-vm
+                    // If the input is an object and not any ArrayBuffer
+                    // or an ArrayBuffer view (this includes all typed arrays and DataViews)
+                    // turn the object into a JSON string, because we suspect
+                    // this is a project.json as an object
+                    // validate expects a string or buffer as input
+                    // TODO not sure if we need to check that it also isn't a data view
+                    input = JSON.stringify(input);
+                }
+                parser(projectAsset.data, false, (err, projectData) => {
```

Comment:
You should handle the error here. Probably the right thing to do is just log it and return, since this is solely so the extension panel can load.  This way we don't have to e.g. validate the project ID, we can just not crash if the projectID is bad.

Reply:
Fixed by 456e99f, but GitHub didn't mark this comment as outdated.